cmd/rpcdaemon/commands/eth_api.go

```diff
@@ -154,11 +154,14 @@ func (api *BaseAPI) blockByHashWithSenders(tx kv.Tx, hash common.Hash) (*types.B
 			return it.(*types.Block), nil
 		}
 	}
-	number := rawdb.ReadHeaderNumber(tx, hash)
-	if number == nil {
+	header, err := api._blockReader.HeaderByHash(context.Background(), tx, hash)
```

Comment:
1. this is de-optimization: read header is slower than read headerNumber (more requests, need parse rlp, allocate header object, etc...)
2. we fill `kv.HeaderNumber` table for all blocks - even for blocks in snapshots. see `eth/stagedsync/stage_snapshots.go: Load(tx, kv.HeaderNumber,`. it means we don't need `_blockReader` to get block number by block hash.

Reply:
3. we store this index in db - because this index is small enough.